refract-spec.md

```diff
@@ -117,29 +485,570 @@ This example will show a JSON object represented as both a normal Refract Elemen
+        "content": "bar"
+      }
+    }
+  ]
+}
+```
+
+Because Object Types can also take elements, if the `foo` element were defined, it could be used as such:
+
+```json
+{
+  "element": "object",
+  "content": [
+    {
+        "element": "foo",
+        "content": "bar"
+    }
+  ]
+}
+```
+
+### Member Element (Element)
+
+The Member Element provides an element structure for the key/value pair type Member Type.
+
+#### Properites
+
+- `element` member (string, fixed)
+- `content` (Member Type)
+
+### Key Element (Element)
+
+The Key Element is used by the Member Type to provide an element for defining keys and meta data and attributes around those keys.
+
+#### Properites
+
+- `element` key (string, fixed)
+- `content` (enum)
+    - (string)
+    - (Element)
+
+### Value Element (Element)
+
+The Value Element is used by the Member Type to provide an element for defining values of keys along with meta data and attributes around those values.
+
+#### Properites
+
+- `element` value (string, fixed)
+- `content` (enum)
+    - (string)
+    - (Element)
+
+## Referencing Elements
+
+The referencing elements are provided as part of the base specification for the purpose of identifying and referencing elements.
+
+### ID Element (Element)
+
+The `id` element is a way to identify an element in a document. IDs MUST be unique throughout a document.
+
+#### Properties
+
+- `element` id (string, fixed)
+- `content` (string) - Unique identifier
+
+#### Example
+
+The `id` element has a string as its content and MAY be used in the following way.
+
+```json
+{
+  "element": "id",
+  "content": "my-id"
+}
+```
+
+The `id` element MAY be used to provide a unique identifer for an element. It is used here in the meta attributes to give an identifier of `foo` to this specific instance of a string with the value of "bar."
+
+```json
+{
+  "element": "string",
+  "meta": {
+    "id": "foo"
+  },
   "content": "bar"
 }
 ```
 
-##### Compact Element Representation
+### Ref Element (Element)
+
+The `ref` element MAY be used to reference elements in remote documents or elements in the local document. The `ref` element transcludes the contents of the element into the document in which it is referenced.
+
+#### Properties
+
+- `element` ref (string, fixed)
+- `content` (Namespace, required) - A URL or fragment identifier. If the content is an absolute URL with a fragment identifier, the fragment identifer references the ID in the corresponding document.
+
+
+#### Examples
+
+Elements MAY be referenced in remote or local documents.
+
+##### Referencing Remote Element
+
+```json
+{
+  "element": "ref",
+  "content": "http://example.com/document#foo"
+}
+```
+
+##### Referencing Local Elements
+
+```json
+{
+  "element": "ref",
+  "content": "foo"
+}
+```
+
+##### Referencing Elements in Prefixed Namespace
+
+This references the element with the ID of `foo` in the prefixed namespace of `ns`.
+
+```json
+{
+  "element": "ref",
+  "content": {
```

Comment:
So, I have been thinking that this is what we have called a namespace element. Frankly, I think in principle it is more of a reference without the idea of transclusion. Almost a `type` and that it has a different connotation as the content of a ref that transcludes its content and a `namespaces` element that 'transcludes' its constraints. Not sure if I have brought this up, but I think we can find a way to collapse and re-use this thing. Then we get some re-use. Just a matter of describing this.